### PR TITLE
polygon/bridge: Set correct time limit

### LINKED ISSUE #11325: astrid bridge creates maps for incorrect block
On BorHeimdall stage:
Screenshot 2024-07-25 at 4 46 29 PM

On Astrid:
Screenshot 2024-07-25 at 4 46 36 PM

### PR DESCRIPTION
**Pull request by shohamc1**
Sync events time limit was not following Indore rules

Fixes #11325

### CODE REVIEW

File: polygon/bridge/bridge.go [hunk 1]
```diff
@@ -126,19 +126,30 @@ func (b *Bridge) Close() {
...
 		}
 
-		blockTimestamp := time.Unix(int64(block.Time()), 0)
-		lastDBID, err := b.store.GetSprintLastEventID(ctx, b.lastProcessedEventID, blockTimestamp, b.stateReceiverABI)
+		var timeLimit time.Time
+		if b.borConfig.IsIndore(block.NumberU64()) {
+			stateSyncDelay := b.borConfig.CalculateStateSyncDelay(block.NumberU64())
+			timeLimit = time.Unix(int64(block.Time()-stateSyncDelay), 0)
+		} else {
+			timeLimit = prevSprintTime
```
Comment (antonis19) [inline polygon/bridge/bridge.go:142]: Isn't prevSprintTime uninitialized at this point?

Comment (shohamc1) [reply polygon/bridge/bridge.go:142]: It is initialized to its zero value which is 0 (and fine in this case)

Comment (antonis19) [reply polygon/bridge/bridge.go:142]: Yeah, so is a timeLimit of 0 correct for this case?

Comment (shohamc1) [reply polygon/bridge/bridge.go:142]: Yes, it produces the correct maps - there are no state sync events for the first sprint.

https://github.com/erigontech/erigon/blob/7566807113d7646878c6cd302e3b18ef2b32ccfe/polygon/bor/bor.go#L1516

**Review by antonis19 — APPROVED** [review]

**Pull request merged**